### PR TITLE
UI test for app navigation and Android CI update

### PR DESCRIPTION
**Pull request by Quinstedt**
The project has been change to a Kotlin project but all Java classes are still there and working. Kotlin has been used to create UI test with espresso.  ControlPad and Joystick have now a composition to the BrokerConnection class ( no longer inheritance)

The test are testing the navigation of the app for what we have in the main.
Test for the leaderboard will be created later when the branch has been merged. ( There only one test att has been commented out related to the leaderboard)
 
All test are passing when running them in the android studio, for both Nexus 6 API 29 and API 31. 
And the android CI has been updated to run instrumeted test. 

Created new comments for important method that should have been send in  #47.

Pull Request #54 has been merge with this one. The bug in the swipe function of the leaderboard has been fixed.

Closes related issue: #42
And closes an issue that has been closed by other team members before but without meeting the acceptance criteria. 
Closes related issue: #30

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

**Comment by platisd** [comment]
Very promising :+1:

**Comment by Quinstedt** [comment]
Referring the comment on the code to this pull request

**Comment by Quinstedt** [comment]
Improvement to this pull request before its merge:

- [ ] Add test for the Leaderboard 
- [ ] Passing test for the Leaderboard

**Comment by platisd** [comment]
Do the tests pass when you run them locally?

**Comment by faisal1201** [comment]
I executed them on Pixel 3 API 30 emulator, all the tests are passing
image

**Comment by faisal1201** [comment]
Before I approve and merge this branch, I'll wait for the leaderboard test to be committed. 
 Good job @Quinstedt .

**Comment by Quinstedt** [comment]
> Do the tests pass when you run them locally?


Yes and they were all passing here as well before I inserted the comment on the improvements that the pull request is waiting for. 

But I will look into it and I hope to close this request today.

**Comment by Quinstedt** [comment]
All test are passing when they are run manually. 
image

**Comment by platisd** [comment]
🤔 Maybe I can take a look later

**Comment by platisd** [comment]
Can't really tell to be honest, I'd need to spend more time on this and I don't have it. I suggest this: If you cannot make it work I can disable these jobs from "voting" in the Pull request.
Instead, you should add a DoD criterion about "running manually the instrumented tests" or something. What do you think?

**Comment by Quinstedt** [comment]
> Can't really tell to be honest, I'd need to spend more time on this and I don't have it. I suggest this: If you cannot make it work I can disable these jobs from "voting" in the Pull request. Instead, you should add a DoD criterion about "running manually the instrumented tests" or something. What do you think?

I can try to look at it  otherwise I can look into creating a DoD and contact you. 
Think the problem is the toastMessage check, I will try without it and see if it works. Otherwise I will create a DoD for that particular part. 
Thank for the help

**Comment by Quinstedt** [comment]
Hi @platisd , it seems like it is working now, there is a problem with the toastMessage check in the CI when I commented the toastCheck the CI is working passing. 

So what do you think?
Should I create a DoD for that toastMessage test? and require a a picture of the passing test without the toastCheck as a comment. 

or should I just comment that part of the code when pushing to the master. 
image

**Comment by Quinstedt** [comment]
Example of the DoD for the toastMessageTest

image

**Comment by platisd** [comment]
@Quinstedt Yes you can add it to DoD :+1: 
Instead of commenting it out, can you instead exclude it from the CI job run? Maybe look [here](https://stackoverflow.com/questions/38572671/exclude-gradle-largetest) by tagging it as a "LargeTest" or something equivalent.

**Comment by Quinstedt** [comment]
@platisd Hi, What should I do to get the added check boxes to the Dod in the pull Request description?
image

File: .github/dod.yaml
Comment (platisd) [inline .github/dod.yaml:11]: These will take effect starting from the next PR.
However, it feels like that all these points should be once instead?

Comment (platisd) [reply .github/dod.yaml:11]: Maybe you can try adding `\n` in the text and see how it will be rendered in the end. Unfortunately you need to merge first to see the results, but doesn't hurt to try. I can approve things quickly if need be.

Comment (Quinstedt) [reply .github/dod.yaml:11]: image
Like this?

Comment (platisd) [reply .github/dod.yaml:11]: Yeap :+1:

Comment (Quinstedt) [reply .github/dod.yaml]: could you please merge this pull request ? =)

**Review by platisd — APPROVED** [review]
I guess the changes are fine with the rest of @DIT113-V22/group-08

**Pull request merged**
